File: ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/Hotel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReservaDeHotel.Models;
public class Hotel
{
    [Key]
    public int IDHotel { get; set; }
    public string? Nome { get; set; }
    public string? Endereco { get; set; }
    public List<Avaliacao>? ListaAvaliacoes { get; set;}
    public string? Descricao { get; set; }
    public string? ListaDeQuartos { get; set; }
    public double AvaliacaoMedia { get; set; }
    public int NumeroTotalDeQuartos { get; set; }
    public int NumeroDeQuartosDisponiveis { get; set; }

    [ForeignKey("Dono")]
    public int IdDono { get; set; } // aqui teria que usar uma ForeignKey
    public Dono Dono {get; set;}
}

File: ReservaHotelFront/src/app/Hotel.ts
import { Avaliacao } from './Avaliacao';
import { Dono } from './Dono';

export class Hotel {
    idHotel: number = 0;
    nome: string = '';
    endereco: string = '';
    avaliacoes: Avaliacao | undefined;
    descricao: string = '';
    listaDeQuartos: string = '';
    avaliacaoMedia: number = 0;
    donoId: number = 0;
    dono: Dono | undefined;
    numeroTotalDeQuartos: number = 0;
    numeroDeQuartosDisponiveis: number = 0;
}
